apps/dashboard/src/components/number-input-with-select.tsx

```diff
@@ -0,0 +1,96 @@
+import { cn } from '@/utils/ui';
+import { FormControl, FormField, FormItem, FormMessagePure } from '@/components/primitives/form/form';
+import { Input } from '@/components/primitives/input';
+import { InputFieldPure } from '@/components/primitives/input';
+import { Select, SelectContent, SelectItem, SelectTrigger, SelectValue } from '@/components/primitives/select';
+import { UseFormReturn } from 'react-hook-form';
+import { useMemo } from 'react';
+
+type InputWithSelectProps = {
+  form: UseFormReturn<any>;
+  inputName: string;
+  selectName: string;
+  options: string[];
+  defaultOption?: string;
+  className?: string;
+  placeholder?: string;
+  isReadOnly?: boolean;
+};
+
+export const NumberInputWithSelect = (props: InputWithSelectProps) => {
```

Comment:
I did this as a component for now - with some effort it could be probably more generic, but its hard to do abstractions now since I'm not sure how or if will the digest reuse this.